cli/pkg/local/app.go

```diff
@@ -182,9 +184,12 @@ func NewApp(ctx context.Context, opts *AppOptions) (*App, error) {
 		"rill.download_row_limit": "0", // 0 means unlimited
 		"rill.stage_changes":      "false",
 	}
-	for k, v := range opts.Variables {
-		vars[k] = v
+	maps.Copy(vars, opts.Variables)
+	envMap, err := godotenv.Read(filepath.Join(projectPath, ".env"))
+	if err != nil && !errors.Is(err, os.ErrNotExist) {
+		return nil, err
 	}
+	maps.Copy(vars, envMap)
```

Comment:
This has some problems:
- Only works on local, not in cloud (`.env` is usually gitignored, but people may have reasons not to, e.g. if using it for non-sensitive variables)
- Duplicates `.env` parsing logic from the parser
- Does not reload on controller restart

I think this can be addressed by:
- Implementing it before starting the controller (in `runtime/registry.go` around line 419) – then it works also in cloud, and also for controller restarts. 
- Implementing a function `rillv1.ParseDotEnv` that only parses the .env file, similar to `rillv1.ParseRillYAML`

Reply:
The idea behind these changes was to set variables in an instance during application boot up on local. For all other cases i.e. cloud and controller restart on local, the env variables should already be set in instance in function `reconcileProjectConfig`.
Alternatively what you are suggesting can also be done.

Reply:
Got it, it's a good point that in subsequent runs it will already have been updated. 

Still wondering about these two issues:
1. On the first run in cloud, there will be a flash of errors like we currently see on local. It would be nice to avoid that, especially if we start sending alerts for project errors.
2. We should try to avoid a controller restart right after startup – it causes a bunch of cancels/reruns of reconcilers, which even if safe, can still seem weird. So maybe we should also incorporate the logic for variables from `rill.yaml` into this code?